### PR TITLE
sync: Send already connected peers to new subscribers

### PR DESCRIPTION
**Pull request by dmitry-markin**
Introduce `SyncEvent::InitialPeers` message sent to new subscribers to allow them correctly tracking sync peers. This resolves a race condition described in https://github.com/paritytech/polkadot-sdk/issues/6573#issuecomment-2563091343.

Fixes https://github.com/paritytech/polkadot-sdk/issues/6573.

### CODE REVIEW

File: prdoc/pr_7011.prdoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Fixes https://github.com/paritytech/polkadot-sdk/issues/6573.
+crates:
+- name: sc-network-gossip
+  bump: major
+- name: sc-network-statement
+  bump: major
+- name: sc-network-sync
+  bump: major
+- name: sc-network-transactions
+  bump: major
```
Comment (bkchr) [inline prdoc/pr_7011.prdoc:16]:
```suggestion
- name: sc-network-statement
  bump: patch
- name: sc-network-sync
  bump: major
- name: sc-network-transactions
  bump: patch
```

**Review by bkchr — APPROVED** [review]
Nice!

**Review by skunert — APPROVED** [review]

**Review by pepoviola — COMMENTED** [review]
Nice! Thanks @dmitry-markin 🙌

**Pull request merged**